server/api.go

```diff
@@ -145,6 +153,28 @@ func (a *API) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	a.router.ServeHTTP(w, r)
 }
 
+func (a *API) ServeHTTPWithMetrics(w http.ResponseWriter, r *http.Request) {
+	recorder := &StatusRecorder{
+		ResponseWriter: w,
+		Status:         200,
+	}
+	now := time.Now()
+
+	a.router.ServeHTTP(recorder, r)
+
+	elapsed := float64(time.Since(now)) / float64(time.Second)
+
+	var routeMatch mux.RouteMatch
+	a.router.Match(r, &routeMatch)
+	if routeMatch.Route != nil {
```

Comment:
If for some strange reason we didn't match a route, should we still observe it but categorize as "unknown"?

Reply:
Yes, that is the idea

Reply:
This is fixed now.